### DISCUSSION TITLE
ULID Schema columns

### DISCUSSION THREAD

**Discussion by DarkGhostHunter** [laravel/framework#44178] [Ideas]
Since the `HasUlid` trait as landed, it also should include Blueprint methods to create ULID columns:

- `foreignUlid()`
- `nullableUlidMorphs()`
- `ulidMorphs()`
- `ulid()`

Also, `foreignIdFor()` should be updated to check for Models that are using ULIDs to create the appropiate column type (`char:26`);

**Comment by Sophist-UK** [comment]
Have you checked the code to see if these exist? If so is there a documentation issue instead?

If so, can you create PRs to address them?

**Comment by FarhanShares** [comment]
<img width="1044" alt="Screenshot 2023-02-07 at 10 27 57 AM" src="https://user-images.githubusercontent.com/3446817/217148395-5168343d-e033-4db4-b443-d87a9aeccabb.png">

As of Laravel 9.50.1, the foreignIdFor() looks like the above. Which is failing my migration with Postgres (not sure with other database right now). It adds 'uuid' column instead 'ulid or char(26)' column.

**Comment by crynobone** [comment]
This should already available in latest Laravel 10.
